pkg/syncer/storagepool/util.go

```diff
@@ -72,6 +72,28 @@ func getDatastoreProperties(ctx context.Context, d *cnsvsphere.DatastoreInfo) (
 	return capacity, freeSpace, ds.Summary.Url, spTypePrefix + dsType, accessible, inMM
 }
 
+// getHostMoIDToK8sNameMap looks up the hostMoid annotation on each k8s node.
+func getHostMoIDToK8sNameMap(ctx context.Context) (map[string]string, error) {
+	log := logger.GetLogger(ctx)
+	hostMoIDTok8sName := make(map[string]string)
+	clientSet, err := k8s.NewClient(ctx)
+	if err != nil {
+		log.Errorf("Failed to create k8s client for cluster, err=%+v", err)
+		return hostMoIDTok8sName, err
+	}
+	nodeList, err := clientSet.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
```

Comment:
How often do you expect `getHostMoIDToK8sNameMap` to be called? `List`ing from the API server can get quite expensive at scale so we tend to use informers if the frequency is high

Reply:
Good point. Allow me to leave this as is since I intend to decouple storagepool controller out of syncer soon. During that refactoring, storagepool would get its own informer. For now, I will add a TODO comment here so that this point does not get lost.